### DISCUSSION TITLE
How to configure semantic commit types?

### DISCUSSION THREAD

**Discussion by ryancausey** [renovatebot/renovate#14026] [Request Help] [answered]
I've noticed recently that renovate has started switching from using `chore(deps)` to `fix(deps)` when updating python packages using Poetry. This is occurring even for patch version bumps, like django 2.2.26 to django 2.2.27. In some cases this behavior is desirable, for example when the django app that is deployed gets a dependency update. However, for library packages where this is an update to the library's lockfile only, and does not effect the version range of django supported by the library, this is not desirable.

I've tried looking at how to customize this behavior to work differently in these two cases, but I'm not finding clear answers in the documentation, but I might be missing something. Does anyone know how to achieve these customizations?

My renovate.json looks like this:

```json
{
    "$schema": "https://docs.renovatebot.com/renovate-schema.json",
    "extends": [
        "config:base"
    ],
    "rangeStrategy": "update-lockfile"
}
```

**Comment by rarkins** [comment, marked as answer]
You are probably seeing the effects of the `:semanticPrefixFixDepsChoreOthers` preset, which is included in `config:base`. If you add that to `ignorePresets` then you should see everything revert back to `chore`.

You can also manually override things using packageRules, e.g.

```
{
  "packageRules": [
    {
      "matchPackagePatterns": [
        "Abc"
      ],
      "semanticCommitType": "chore"
    }
  ]
}
```

**Reply by ryancausey** [reply]
@rarkins thanks for the info!

Is there some way to hook into whether the acceptable range of packages is changing vs a lockfile only update to set the semantic commit type?

For example:

If renovate only modifies `poetry.lock` to bump Django 2.2.26 to 2.2.27, set that to `chore(deps)`. However, if renovate updated the `pyproject.toml` range from "django > 2.1, django < 3.0" to "django > 3.0, django < 4.0" and modifies `poetry.lock` to bump Django 2.2.27 to Django 3.2.2, set that to `fix(deps)`.

**Reply by rarkins** [reply]
No, there is not. I recommend you pin dependencies and then use patch/minor/major rules.

**Reply by ryancausey** [reply]
I'm assuming you're talking about using this matching config option to be able to set the [semanticCommitType ](https://docs.renovatebot.com/configuration-options/#semanticcommittype) based on the update type: https://docs.renovatebot.com/configuration-options/#matchupdatetypes
